Mathlib/Algebra/Module/Submodule/LinearMap.lean

```diff
@@ -0,0 +1,476 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+
+import Mathlib.Algebra.Module.Equiv
+import Mathlib.Algebra.Module.Submodule.Basic
+
+#align_import algebra.module.submodule.basic from "leanprover-community/mathlib"@"8130e5155d637db35907c272de9aec9dc851c03a"
+
+/-!
+
+# Linear maps involving submodules of a module
+
+In this file we define a number of linear maps involving submodules of a module.
```

Comment:
Could you add a list of important declarations in this file? I would nominate `subtype`, `domRestrict`, `restrict` and `ofLe`.